TakingITGlobal/commit2act
# [TIG-824] Placeholder pages on routes and initial data fetching

## Ticket

* [TIG-824](https://app.clickup.com/t/9009201449/TIG-824)

## Description

This PR primarily adds dedicated (placeholder) pages for each sub-route within the Log Action flow. The first page loads and renders real actions data, including links to the details page for the selected action. The second page loads the action items and selected action's details (to render the name above the main heading). 

The remaining placeholder pages have no data requests at this time. They're all pretty basic - just the page and an associated test file. Hence the larger number of files on this one!

## Testing

1. Pull down and install the branch (`npm install`)
2. Run the test suite and confirm it passes (`npm run test:basic`)
3. Run the local development app (`npm run start:dev`)
4. Login and use the dashboard or main nav link for Log Action
5. Confirm the 6 action types load within the page with images and text for each
6. Click/select any of the action links to visit the Enter Details page
7. Confirm the action name renders above the "Add Action" heading
8. Confirm the action items are logged in the console when loaded
9. Visit `/log-action/1/create` and confirm the "Nice work" heading renders
10. Visit `/log-action/1/2` and confirm the "Points or CO2" heading renders
11. Visit `/log-action/1/2/bonus` and confirm the "Bonus question" heading renders
12. Visit `/log-action/1/2/share` and confirm the "Share on social" heading renders

## Screenshots

**Brief video demo of landing on the Select Action page, selecting one, and reaching the Enter Details placeholder page with live data:**

https://github.com/TakingITGlobal/commit2act/assets/1672105/268836ec-6aa2-46bc-8eea-08025a4f92ae
